straxen/plugins/acqmon_processing.py

```diff
@@ -130,4 +133,82 @@ def merge_vetos(channels, gap, dtype, t):
 @numba.njit
 def channel_select(rr, ch_stop, ch_start):
     """Return data in between stop and start channels of the acquisition monitor (AM)"""
-    return rr[((rr['channel'] == ch_stop) | (rr['channel'] == ch_start))] 
+    return rr[((rr['channel'] == ch_stop) | (rr['channel'] == ch_start))]
+
+
+
+class VetoProximity(strax.Plugin):
+    """ 
+    Find the closest next/previous veto start and end to each event center.
+    
+    previous_x: Time in ns between the time center of an event and the previous x
+    next_x: Time in ns between the time center of an event and the next x
+    This also considers any x inside the event. x could be either:
+        - busy_x: busy on/off signal
+        - he_x:   high energy channels busy on/off signal
+        - hev_x:  high energy veto on/off signal
+    """
+        
+    __version__ = '0.0.1'
+    depends_on = ('veto_intervals','events') 
+    provides  = ('veto_proximity')
+    data_kind = ('veto_proximity')
+    
+    veto_names = ['busy', 'he', 'hev']
+    
+    def infer_dtype(self):
+        dtype = []
+        for n in self.veto_names:
+            dtype.append((('Time to previous %s veto start [ns]' %n, 'previous_%s_on' %n), np.int64))
+            dtype.append((('Time to previous %s veto end [ns]' %n, 'previous_%s_off' %n), np.int64))
+            dtype.append((('Time to next %s veto start [ns]' %n, 'next_%s_on' %n), np.int64))
+            dtype.append((('Time to next %s veto end [ns]' %n, 'next_%s_off' %n), np.int64))
+        
+        dtype += strax.time_fields
+
+        return dtype
+    
+    def setup(self):
+        self.states = ['on', 'off']
+        
+    def compute(self, events, veto_intervals):
+        res = defaultdict(list)
+        
+        t_event_centers = (events['time'] + events['endtime'])//2
```

Comment:
I can imagine that you want to have the end/start depending on if you have ` Time to previous ` or `Time to next ` . An event is just a bunch of peaks where the s1 and s2 may be very far apart in time. So I would have expected that this center time might not be what you need based on your description.

Reply:
> I think we are not on the same page yet, maybe better to move to slack :D

Sure :)

Reply:
The resulting output from this plugin would be used in a cut plugin, which will look something like what we had in lax in XENON1T, with the "nearest_x" calculated there from the "previous_x" and "next_x" (or I might add this to this plugin). 
https://github.com/XENON1T/lax/blob/1ac8e2625d82d3fecce149f8b1e9b46959cf00f8/lax/lichens/sciencerun0.py#L147

Regarding "save_when = strax.SaveWhen.NEVER", yeah I think it is a good idea.